metrotranscom/doorway
# feat: add listing search to public site

# Pull Request Template

## Issue Overview


## Description

This was a bit of a rush, so I apologize in advance that it's a bit cluttered.

This PR adds a wireframe search/filter box that queries the backend server and displays the results.  Right now it supports county, bedrooms, bathrooms, and rent.  It does not have the SVG map, but I'm assuming that will happen later.  The "clear" button doesn't do anything right now, but the rest of the functionality is there.  The search page is under `/search-test` rather than `/listings` to keep from interfering with any other ongoing work.

I also added some additional classes for building query filters and to abstract away calls to the listings API.  I tried to clean up any loose ends and remove duplicate functions where possible so there wouldn't be competing implementations.

## How Can This Be Tested/Reviewed?

This branch is up to date with main, so you should be able to test off of listings in your local DB.  Just make sure you set `CORS_ORIGINS=http://localhost:3000` on your `backend/core` .env file so the requests don't get blocked.  The public site also now expects `LISTINGS_QUERY` to be set to `/listings`; it automatically adds the `/combined` suffix specifically and solely for searches.

![image](https://github.com/metrotranscom/doorway/assets/23032560/bb1eddd9-6fb3-45cb-a366-1c5fc9e43091)

This PR passes all current unit tests but does not add any new ones at this point other than for the search string builder/parser.  The wireframe pieces will need to be replaced with the properly styled components, so any UI tests are best saved for then.

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
